nixos/modules/installer/cd-dvd/iso-image.nix

```diff
@@ -901,6 +899,44 @@ in
       "overlay"
     ];
 
+    boot.initrd.systemd = lib.mkIf config.boot.initrd.systemd.enable {
+      emergencyAccess = true;
+
+      # Most of util-linux is not included by default.
+      initrdBin = [ config.boot.initrd.systemd.package.util-linux ];
+      services.copytoram = {
+        description = "Copy ISO contents to RAM";
+        requiredBy = [ "initrd.target" ];
+        before = [
+          "${utils.escapeSystemdPath "/sysroot/nix/.ro-store"}.mount"
+          "initrd-switch-root.target"
+        ];
+        unitConfig = {
+          RequiresMountsFor = "/sysroot/iso";
+          ConditionKernelCommandLine = "copytoram";
+        };
+        serviceConfig = {
+          Type = "oneshot";
+          RemainAfterExit = true;
+        };
+        path = [
+          pkgs.coreutils
+          config.boot.initrd.systemd.package.util-linux
+        ];
+        script = ''
+          device=$(findmnt -n -o SOURCE --target /sysroot/iso)
+          fsSize=$(blockdev --getsize64 "$device" || stat -Lc '%s' "$device")
+          mkdir -p /tmp-iso
+          mount --bind --make-private /sysroot/iso /tmp-iso
+          umount /sysroot/iso
+          mount -t tmpfs -o size="$fsSize" tmpfs /sysroot/iso
+          cp -r /tmp-iso/* /sysroot/iso/
+          umount /tmp-iso
+          rm -r /tmp-iso
+        '';
+      };
+    };
```

Comment:
(This is kind of gross, but I realize that doing anything about it is out of scope for this PR.)